src/argilla/client/nightly.py

```diff
@@ -0,0 +1,367 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import sys
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+if sys.version_info > (3, 7):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+from pydantic import BaseModel, Field, create_model, validator
+from tqdm import tqdm
+
+import argilla as rg
+
+if TYPE_CHECKING:
+    from argilla.client.api import Argilla
+    from argilla.client.sdk.datasets.v1.models import (
+        FeedbackDatasetModel,
+        FeedbackItemModel,
+    )
+
+
+class ValueSchema(BaseModel):
+    value: Union[str, int]
+
+
+class ResponseSchema(BaseModel):
+    values: Dict[str, ValueSchema]
+    status: Literal["submitted", "missing", "discarded"]
+
+
+class FeedbackRecord(BaseModel):
+    fields: Dict[str, str]
+    response: Optional[ResponseSchema] = None
+    external_id: Optional[str] = None
+
+    @validator("response", always=True)
+    def response_must_have_values(cls, v):
+        if not v:
+            v = ResponseSchema(values={}, status="submitted")
+        return v
+
+
+class FieldSchema(BaseModel):
+    name: str
+    title: Optional[str] = None
+    required: Optional[bool] = True
+    settings: Dict[str, Any]
+
+    @validator("title", always=True)
+    def title_must_have_value(cls, v, values):
+        if not v:
+            return values["name"].capitalize()
+        return v
+
+
+class TextField(FieldSchema):
+    settings: Dict[str, Any] = Field({"type": "text"}, const=True)
+
+
+class QuestionSchema(BaseModel):
+    name: str
+    title: Optional[str] = None
+    description: Optional[str] = None
+    required: Optional[bool] = True
+    settings: Dict[str, Any]
+
+    @validator("title", always=True)
+    def title_must_have_value(cls, v, values):
+        if not v:
+            return values["name"].capitalize()
+        return v
+
+
+class TextQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field({"type": "text"}, const=True)
+
+
+class RatingQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field({"type": "rating"})
+    values: List[int]
+
+    @validator("values", always=True)
+    def update_settings_with_values(cls, v, values):
+        if v:
+            values["settings"]["options"] = [{"value": value} for value in v]
+        return v
+
+
+class FeedbackDataset:
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        *,
+        workspace: Optional[Union[rg.Workspace, str]] = None,
+        id: Optional[str] = None,
+    ) -> None:
+        self.client: "Argilla" = rg.active_client()
```

Comment:
I prefer to let users work locally with these feedback datasets, and do explicit interactions with the argilla server, as `dataset.save` or `dataset.push`

Similar for loading. We should load datasets from argilla with an explicit method.
````
python
ds = FeedbackDataset.from_name_and_workspace(name, workspace)
...
````

Reply:
Yes, I was already preparing `push_to_argilla` and `from_argilla` as we discussed! 😄